media/stitcher/src/test/java/com/example/stitcher/VodSessionTest.java

```diff
@@ -0,0 +1,184 @@
+    return varValue;
+  }
+
+  @BeforeClass
+  public static void checkRequirements() {
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    PROJECT_ID = requireEnvVar("GOOGLE_CLOUD_PROJECT");
+  }
+
+  @Before
+  public void beforeTest()
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    TestUtils.cleanStaleVodConfigs(PROJECT_ID, TestUtils.LOCATION);
+    originalOut = System.out;
+    bout = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(bout));
+
+    VOD_CONFIG_NAME =
+        String.format("locations/%s/vodConfigs/%s", TestUtils.LOCATION, VOD_CONFIG_ID);
+    VodConfig configResponse =
+        CreateVodConfig.createVodConfig(
+            PROJECT_ID,
+            TestUtils.LOCATION,
+            VOD_CONFIG_ID,
+            TestUtils.VOD_URI,
+            TestUtils.VOD_AD_TAG_URI);
+    assertThat(configResponse.getName(), containsString(VOD_CONFIG_NAME));
+
+    // Session IDs are autogenerated.
+    VOD_SESSION_NAME_PREFIX = String.format("locations/%s/vodSessions/", TestUtils.LOCATION);
+    VodSession sessionResponse =
+        CreateVodSession.createVodSession(PROJECT_ID, TestUtils.LOCATION, VOD_CONFIG_ID);
+    assertThat(sessionResponse.getName(), containsString(VOD_SESSION_NAME_PREFIX));
+
+    VOD_SESSION_NAME = sessionResponse.getName();
+    String[] id = VOD_SESSION_NAME.split("/");
+    SESSION_ID = id[id.length - 1];
+
+    ListVodAdTagDetailsPagedResponse adtagResponse =
+        ListVodAdTagDetails.listVodAdTagDetails(PROJECT_ID, TestUtils.LOCATION, SESSION_ID);
+    for (VodAdTagDetail vodAdTagDetail : adtagResponse.iterateAll()) {
+      AD_TAG_DETAIL_NAME = vodAdTagDetail.getName();
+    }
+    id = AD_TAG_DETAIL_NAME.split("/");
+    AD_TAG_DETAIL_ID = id[id.length - 1];
+
+    ListVodStitchDetailsPagedResponse stitchResponse =
+        ListVodStitchDetails.listVodStitchDetails(PROJECT_ID, TestUtils.LOCATION, SESSION_ID);
+    for (VodStitchDetail vodStitchDetail : stitchResponse.iterateAll()) {
+      STITCH_DETAIL_NAME = vodStitchDetail.getName();
+    }
+    id = STITCH_DETAIL_NAME.split("/");
+    STITCH_DETAIL_ID = id[id.length - 1];
+
+    bout.reset();
+  }
+
+  @Test
+  public void testGetVodSession() throws IOException {
+    VodSession response = GetVodSession.getVodSession(PROJECT_ID, TestUtils.LOCATION, SESSION_ID);
+    assertThat(response.getName(), containsString(VOD_SESSION_NAME));
+    bout.reset();
+  }
+
+  @Test
+  public void testListVodAdTagDetailsTest() throws IOException {
+    ListVodAdTagDetailsPagedResponse response =
+        ListVodAdTagDetails.listVodAdTagDetails(PROJECT_ID, TestUtils.LOCATION, SESSION_ID);
+    Boolean pass = false;
+    for (VodAdTagDetail vodAdTagDetail : response.iterateAll()) {
+      if (vodAdTagDetail.getName().contains(VOD_SESSION_NAME.concat("/vodAdTagDetails/"))) {
+        pass = true;
+      }
+    }
+    assert (pass);
```

Comment:
You can directly assert if a list contains an element. Something like:

` assertThat(response.iterateAll()).contains(VOD_SESSION_NAME.concat("/vodAdTagDetails/"));`

Reply:
Applies to all list tests.

Reply:
on a second look,`response.iterateAll()` doesn't return a list of primitives. So this wouldn't be possible.

Let's keep the original approach, but add a `break;` statement in line 136 after `pass = true;` . Applies to all list samples

Reply:
Done